# Influenza Forecasting using ILINet Data

## Project Overview

This notebook focuses on understanding the ILINet dataset before any preprocessing or modeling.

### Objectives

- Load the dataset
- Inspect its structure
- Examine data types
- Identify missing values
- Check for duplicates
- Generate descriptive statistics
- Document observations

#### 1. Import Libraries

In [ ]:
# Import libraries
import warnings

import pandas as pd

from src.data_loader import load_ilinet

warnings.filterwarnings("ignore")

#### 2. Display Settings

In [2]:
# Display settings
pd.set_option("display.max_columns", None)
pd.set_option("display.max_rows", 20)
pd.set_option("display.width", 120)

#### 3. Load Dataset

In [3]:
# Load dataset
df = load_ilinet()

#### 4. Dataset Overview

In [4]:
# Check first 5 rows
df.head()

,REGION TYPE,REGION,YEAR,WEEK,% WEIGHTED ILI,%UNWEIGHTED ILI,AGE 0-4,AGE 25-49,AGE 25-64,AGE 5-24,AGE 50-64,AGE 65,ILITOTAL,NUM. OF PROVIDERS,TOTAL PATIENTS
0,National,X,1997,40,1.10148,1.21686,179,X,157,205,X,29,570,192,46842
1,National,X,1997,41,1.20007,1.28064,199,X,151,242,X,23,615,191,48023
2,National,X,1997,42,1.37876,1.23906,228,X,153,266,X,34,681,219,54961
3,National,X,1997,43,1.19920,1.14473,188,X,193,236,X,36,653,213,57044
4,National,X,1997,44,1.65618,1.26112,217,X,162,280,X,41,700,213,55506


In [5]:
# Check last 5 rows
df.tail()

,REGION TYPE,REGION,YEAR,WEEK,% WEIGHTED ILI,%UNWEIGHTED ILI,AGE 0-4,AGE 25-49,AGE 25-64,AGE 5-24,AGE 50-64,AGE 65,ILITOTAL,NUM. OF PROVIDERS,TOTAL PATIENTS
1494,National,X,2026,20,1.58610,1.52377,8003,11158,X,11733,4319,4562,39775,4209,2610298
1495,National,X,2026,21,1.56831,1.48638,7858,10624,X,10221,4139,4768,37610,4191,2530305
1496,National,X,2026,22,1.39393,1.33762,7195,9939,X,8869,3940,4413,34356,4149,2568443
1497,National,X,2026,23,1.23701,1.18346,6552,8728,X,7477,3513,3931,30201,4110,2551931
1498,National,X,2026,24,1.11953,1.08669,6060,7796,X,6654,3183,3528,27221,4072,2504956


In [6]:
# Check dataset structure
print(f"Shape: {df.shape}, (Rows, Columns)")
print("\nData Types:")
df.info()
print("\nSummary Statistics:")
display(df.describe().round(2))

Shape: (1499, 15), (Rows, Columns)

Data Types:
<class 'pandas.DataFrame'>
RangeIndex: 1499 entries, 0 to 1498
Data columns (total 15 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   REGION TYPE        1499 non-null   str    
 1   REGION             1499 non-null   str    
 2   YEAR               1499 non-null   int64  
 3   WEEK               1499 non-null   int64  
 4   % WEIGHTED ILI     1499 non-null   float64
 5   %UNWEIGHTED ILI    1499 non-null   float64
 6   AGE 0-4            1499 non-null   int64  
 7   AGE 25-49          1499 non-null   str    
 8   AGE 25-64          1499 non-null   str    
 9   AGE 5-24           1499 non-null   int64  
 10  AGE 50-64          1499 non-null   str    
 11  AGE 65             1499 non-null   int64  
 12  ILITOTAL           1499 non-null   int64  
 13  NUM. OF PROVIDERS  1499 non-null   int64  
 14  TOTAL PATIENTS     1499 non-null   int64  
dtypes: float64(2), int64(8), str(5)
mem

,YEAR,WEEK,% WEIGHTED ILI,%UNWEIGHTED ILI,AGE 0-4,AGE 5-24,AGE 65,ILITOTAL,NUM. OF PROVIDERS,TOTAL PATIENTS
count,1499.00,1499.00,1499.00,1499.00,1499.00,1499.00,1499.00,1499.00,1499.00,1499.00
mean,2011.61,26.55,1.93,1.87,4726.87,7000.66,1716.11,20398.95,1772.26,848216.90
std,8.31,15.14,1.49,1.42,5919.81,10150.91,3196.69,29764.01,1225.46,779950.17
min,1997.00,1.00,0.00,0.00,0.00,0.00,0.00,0.00,0.00,0.00
25%,2004.00,13.00,0.99,0.97,835.00,1141.50,185.00,3021.50,755.50,219198.00
50%,2012.00,26.00,1.50,1.44,2564.00,3316.00,504.00,9188.00,1512.00,623077.00
75%,2019.00,40.00,2.41,2.34,6235.50,8265.00,1746.50,24867.00,2578.50,1116382.00
max,2026.00,53.00,8.28,8.48,40575.00,76137.00,33315.00,222652.00,4370.00,2890504.00


#### 5. Data Quality Assessment

In [7]:
# Missing values summary
missing_summary = pd.DataFrame({
    "missing_values": df.isna().sum(),
    "percentage (%)": (df.isna().mean() * 100).round(2)
})
display(missing_summary)

,missing_values,percentage (%)
REGION TYPE,0,0.0
REGION,0,0.0
YEAR,0,0.0
WEEK,0,0.0
% WEIGHTED ILI,0,0.0
%UNWEIGHTED ILI,0,0.0
AGE 0-4,0,0.0
AGE 25-49,0,0.0
AGE 25-64,0,0.0
AGE 5-24,0,0.0


In [8]:
# Check for duplicate rows
duplicate_count = df.duplicated().sum()

print(f"Number of duplicate rows: {duplicate_count}")

Number of duplicate rows: 0


In [9]:
# Count occurrences of "X" in each column
x_summary = (pd.DataFrame({
        "count_of_X": (df == "X").sum()
    })
    .query("count_of_X > 0")
    .sort_values("count_of_X", ascending=False)
)

display(x_summary)

,count_of_X
REGION,1499
AGE 25-64,872
AGE 25-49,532
AGE 50-64,532


#### 6. Temporal Coverage

In [10]:
# Weekly records per year
year_counts = (
    df.groupby("YEAR")
      .size()
      .rename("weekly_records")
      .reset_index()
)

display(year_counts)

,YEAR,weekly_records
0,1997,14
1,1998,52
2,1999,52
3,2000,52
4,2001,52
...,...,...
25,2022,52
26,2023,52
27,2024,52
28,2025,53


# Key Findings

The initial inspection of the ILINet dataset provides the following observations:

- The dataset contains **1,499 weekly influenza surveillance records** measured across **15 variables**.
- The data spans **1997 to 2026**, with **1997** and **2026** representing partial years. Most years contain **52 epidemiological weeks**, while **2025** contains **53 weeks**.
- No missing (`NaN`) values or duplicate records were identified.
- Several variables contain `"X"` values that require further investigation before preprocessing and modeling.
- Overall, the dataset appears well structured and is suitable for further cleaning and preparation.

The next notebook focuses on investigating the `"X"` values, correcting data types where necessary, and preparing the dataset for exploratory analysis and forecasting.